kratos/tests/test_container_expression.py

```diff
@@ -665,6 +665,13 @@ def test_VectorWrite(self):
             velocity = Kratos.Array3([vector[i * 3], vector[i * 3 + 1], vector[i * 3 + 2]])
             self.assertVectorAlmostEqual(velocity, self._GetValue(entity, Kratos.VELOCITY))
 
+    def test_GetMaxDepth(self):
+        a = Kratos.Expression.LiteralExpression.Create(1, 10)
+        b = a + 10
+        c = b * 2
+        d = c ** 2
+        self.assertEqual(d.GetMaxDepth(), 4)
```

Comment:
please add tests for every expression type, and also for cases where different branches have different depths

Reply:
done :)